### PR TITLE
feat: implement rfftn to Paddle Frontend (#23484)

### LINKED ISSUE #23484: rfftn

### PR DESCRIPTION
**Pull request by Kilvny**
# PR Description
added rfftn to Paddle Frontend and added tests related to it

## Related Issue
 (#23484)

Close #23484 

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you follow the steps we provided?

### CODE REVIEW

**Comment by Kilvny** [comment]
Why these 2 check are failing, can you help me with that?

**Review by KareemMAX — CHANGES_REQUESTED** [review]
Hey @Kilvny, seems like you have edited the demos submodule by mistake. Can you please revert that?

**Comment by Kilvny** [comment]
> Hey @Kilvny, seems like you have edited the demos submodule by mistake. Can you please revert that?

Done @KareemMAX

**Comment by Kilvny** [comment]
@Mr-Niraj-Kulkarni

File: ivy/functional/frontends/paddle/fft.py [hunk 1]
```diff
@@ -131,3 +131,29 @@ def rfftfreq(n, d=1.0, dtype=None, name=None):
...
+)
+@to_ivy_arrays_and_back
+def rfftn(x, s=None, axes=None, norm="backward", name=None):
+    """Compute the N-dimensional discrete Fourier Transform over any number of axes in
+    an M-dimensional real array by means of the Fast Fourier Transform (FFT)."""
+    if s is None:
+        s = x.shape
+
+    # Apply rfft along the last axis
+    rfft_result = ivy.rfftn(x, s=s, axes=axes, norm=norm, name=name)
```
[the remark this answers is not in the document]
Comment (Kilvny) [reply ivy/functional/frontends/paddle/fft.py]: Thank you, the change is done, 
FYA

**Review by YushaArif99 — CHANGES_REQUESTED** [review]
Hi @Kilvny 

Apologies for the delay in review. Have been a little swamped with work recently and so havent had the time to review any PRs 😅 

With regards to the PR itself, the implementation does look good overall. However, I am noticing that the CI tests are failing [here](https://github.com/unifyai/ivy/actions/runs/6235375100/job/16924466164?pr=23537#step:3:200) with the following error: 
```python
E   ivy.utils.exceptions.IvyBackendException: tensorflow: rfftn: Invalid FFT input: `x` must be of a real dtype. Received: <dtype: 'complex128'>
```

Could you kindly look into fixing this and re-request a review once that's done? Thanks!😄

**Comment by Kilvny** [comment]
> [YushaArif99](/YushaArif99)

Hi @YushaArif99 Thank you so much for the reply,
Apologies for taking so long to work on this changes, 
I was pretty busy and I got a new job recently

anyways, I resolved the conflicts, but regarding the checks that fails, I need some help and guidance I don't know exactly what causes the tests to fail, your help is highly appreciated 

@KareemMAX

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**